### PR TITLE
Keep worker status visible under strict CSP

### PR DESCRIPTION
**Pull request by jonastemplestein**
## Why

Protected `internal--<project>` requests are intercepted by OS authentication before they reach the project worker. Those responses use a strict Content Security Policy, and project ingress previously skipped the worker-status overlay whenever CSP was present.

## What changed

- Render the worker-status badge as scriptless declarative shadow DOM with native `details` / `summary` interaction.
- Generate a fresh 144-bit nonce per transformed response and authorize only the injected shadow stylesheet.
- Apply the narrow style-policy rewrite to enforced headers, report-only headers, and CSP meta policies.
- Preserve the existing overlay opt-out and independent default-favicon fallback.
- Cover strict internal-app authentication in the deployed browser suite.

## Security

The transform never adds script permission, `unsafe-inline`, or a host source. When styles inherit from `default-src`, ingress copies the existing fallback sources into a new `style-src` directive and adds the nonce there; the nonce never enters `default-src` or the script fallback. Static overlay metadata is HTML-escaped.

## Verification

- `pnpm typecheck`
- `pnpm lint` (zero warnings)
- 25 focused worker-overlay tests
- Preview browser coverage asserts that the strict-CSP internal-app sign-in shows the overlay, contains no overlay script, and authorizes exactly its stylesheet nonce

---

> [!NOTE]
> **Medium Risk**
> Changes HTML ingress and CSP rewriting on project-app documents, including auth pages with `default-src 'none'`, though the rewrite is narrowly limited to authorizing one style nonce and does not add script sources.
> 
> **Overview**
> **Project ingress no longer skips the worker-status overlay when a page has CSP.** The corner badge is now **scriptless declarative shadow DOM** (`iterate-worker-status`, `data-iterate-worker-overlay`) with native **`details`/`summary`** for the panel instead of `__iterateWorkerOverlay` inline script.
> 
> On CSP-protected HTML, **`applyProjectWorkerOverlay`** mints a per-response style nonce, rewrites **enforced/report-only CSP headers** and **CSP meta** via new **`allowStyleNonceInCsp`** (nonce only on `style-src` / `style-src-elem`, or a derived `style-src` from `default-src`—never widening script), and applies that nonce to the overlay stylesheet. Injection still falls back at document end when there is no `<body>`.
> 
> Docs, unit tests (CSP edge cases), e2e marker updates, and a Playwright check on the strict-CSP **internal** sign-in page assert the overlay is visible with **zero scripts** and a matching CSP nonce.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 59c2ceef2ec07b1df948238badee8253e1f8911d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

| Group | Lines | Significant |
| --- | ---: | ---: |
| Product | +140 -70 | +112 -55 🟩🟩🟩🟥🟥 |
| Tests | +99 -18 | +90 -15 🟩🟩🟩⬜⬜ |
| Docs | +14 -7 | +14 -7 🟩⬜⬜⬜⬜ |
| **Total** | **+253 -95** | **+216 -77** 🟩🟩🟩🟩🟥 |

<sub>Lines counts every changed line; Significant ignores blank lines and JS comments. Between b3dd32c and 59c2cee, bucketed first-match-wins into the groups defined in `scripts/ci/loc-report.ts`.</sub>

## Environment Config Lease

<details>
<summary>No preview slot recorded.</summary>

| app | status | commit | preview | size (gzip) | deploy duration | test duration | retries | cleanup duration | workflow run | updated | summary |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
| Auth | released | `59c2cee` | [https://auth.iterate-preview-2.com](https://auth.iterate-preview-2.com) | 811.3 KiB | 32.8s | 11.6s |  | 2.7s | [Workflow run](https://github.com/iterate/iterate/actions/runs/586861817372584) | 2026-07-21T16:06:20.878Z | Preview app released. |
| Dummy Petshop | released | `59c2cee` | [https://dummy-petshop.iterate-preview-2.com](https://dummy-petshop.iterate-preview-2.com) | 198.2 KiB | 6.1s | 5.7s |  | 581ms | [Workflow run](https://github.com/iterate/iterate/actions/runs/586861817372584) | 2026-07-21T16:06:18.807Z | Preview app released. |
| OS | released | `59c2cee` | [https://os.iterate-preview-2.com](https://os.iterate-preview-2.com) | 4.50 MiB (-11.5 KiB vs main) | 97.2s | 232.9s | 4 retried: catalogue example "stream-cross-post" runs identically across runtimes (vitest x1) — example "stream-cross-post" failed in the run-script runtime: stream-wait-timeout: Timed out waiting for stream event after 10000ms (the public deadline expired while recovery re-armed one-shot waits). · agent create installs only generic machinery; later events configure it (vitest x1) — Durable Object storage operation exceeded timeout which caused object to be reset. · crossPostTo copies matching events with source provenance (vitest x1) — stream-unavailable: Durable Object storage operation exceeded timeout which caused object to be reset. · Worker build pipeline bundles multi-file TypeScript inline sources (vitest x1) — stream-wait-timeout: Timed out waiting for stream event after 60000ms (the public deadline expired while recovery re-armed one-shot waits). | 16.5s | [Workflow run](https://github.com/iterate/iterate/actions/runs/586861817372584) | 2026-07-21T16:06:34.757Z | Preview app released. |

</details>

### CODE REVIEW

File: apps/os/src/domains/workers/worker-serve-overlay.ts [hunk 1]
```diff
@@ -69,6 +69,62 @@ function escapeHtml(text: string): string {
...
+          : names.includes("default-src")
+            ? "default-src"
+            : null;
+      if (target === null) return directives.join("; ");
+      return directives
+        .map((directive, index) => {
+          if (names[index] !== target) return directive;
+          const sources = directive.split(/\s+/u);
+          return sources.includes(nonceSource) ? directive : `${directive} ${nonceSource}`;
+        })
```
[the remark this answers is not in the document]
Comment (jonastemplestein) [reply apps/os/src/domains/workers/worker-serve-overlay.ts]: Fixed in ee10147d5. When styles inherit from `default-src`, the rewriter now copies the effective fallback sources into a new `style-src` directive and adds the overlay nonce there. `default-src` remains unchanged, so an absent `script-src` cannot inherit the platform style nonce. Added a focused regression asserting the nonce never appears in `default-src`; the 25 overlay tests, full typecheck, and lint all pass.
